### DISCUSSION TITLE
Meep Units

### DISCUSSION THREAD

**Discussion by amin0098** [NanoComp/meep#2347] [Q&A] [answered]
Hi,
I'm a bit confused about the meep units. So, in the documentation, a parameter called 'a' is defined as equal to 1-micron length. Since the equations are dimensionless, does it mean that parameter 'a' can be considered to be any number and other parameters such as frequency and resolution can be calculated accordingly or everything is normalized to 1 micron and parameters should be calculated based on the fact that everything is normalized to 1 micron? I'd appreciate it if you clarify that.

**Comment by smartalecH** [comment, marked as answer]
The user defines the length scale. Because Maxwell's equations are scale invariant, everything is self consistent. 

If you assume that "1" corresponds to 1 micron, then a frequency of "1" corresponds to a wavelength of 1 micron. 

Note that you could take the _results_ of the _same_ simulation and implicitly change to lengthscale and wavelength (eg such that 1 corresponds to 1 meter) and everything is exactly the same (except now the wavelength corresponds to 1 meter). The relative fields between the two simulations are identical. Just the interpretation differs. 

In other words, you don't have to tell meep anything about units. You just need to be consistent with how you set things up and interpret the results.

**Reply by amin0098** [reply]
Thank you very much Alec for taking the time to answer me.
